api-schema/xdb.yaml

```diff
@@ -114,6 +114,39 @@ paths:
             application/json:
               schema:
                 $ref: '#/components/schemas/ApiErrorResponse'
+  /api/v1/xdb/service/process-execution/rpc:
+    post:
+      summary: execute a RPC method of a process execution
+      requestBody:
+        content:
+          'application/json':
+            schema:
+              $ref: '#/components/schemas/ProcessExecutionRpcRequest'
+      responses:
+        '200':
+          description: successful operation
+          content:
+            'application/json':
+              schema:
+                $ref: '#/components/schemas/ProcessExecutionRpcResponse'
+        '400':
+          description: invalid request
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ApiErrorResponse'
+        '404':
+          description: Process execution not exists
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ApiErrorResponse'
+        '424':
+          description: global attributes write failure / worker RPC execution failure
```

Comment:
With this direction(I like it), I think we need to add a enum to tell which error type it is, like in iWF: https://github.com/indeedeng/iwf-idl/blob/af1891eb032c534b0b75894c611ead7e2954777d/iwf.yaml#L612

Otherwise, SDK won't be able to differentiate the two types of error.

Reply:
Maybe SDK can just throw the error messages with the same status code?

Reply:
It's not about message. User may need to get status code for different handling. 
E.g. for global attr DB operation error, there should be a Postgres error code to use, in addition to the error message. 

for error from worker execution, it may just be the http status code. 

When you are doing that, I think it's better to rename 
`originalWorkerErrorStatus` to `originalWorkerErrorCode`, and change from integer to string, so that it can be more generic. 

https://github.com/xdblab/xdb-apis/blob/eb0596e22fb87e960cee8b80b8378b0dbaa2b2ea/api-schema/xdb.yaml#L544

Reply:
I am confused about the content you are talking, because I am thinking global attributes updating error is not a worker side error, it's server side.